File: src/main/java/net/lomeli/lomlib/client/patreon/PatreonList.java
package net.lomeli.lomlib.client.patreon;

import com.google.gson.Gson;

import java.io.InputStreamReader;
import java.io.Reader;
import java.net.URL;
import java.util.ArrayList;
import java.util.List;

import net.minecraft.entity.player.EntityPlayer;

import net.lomeli.lomlib.core.Strings;

public class PatreonList {
    private List<String> idList;

    public PatreonList() {
        idList = new ArrayList<String>();
    }

    public void getLatestList() {
        try {
            idList.clear();
            Gson gson = new Gson();
            Reader file = new InputStreamReader(new URL(Strings.PATREON_URL).openStream());
            String[] json = gson.fromJson(file, String[].class);
            file.close();
            if (json != null) {
                for (String s : json) {
                    idList.add(s);
                }
            }
        } catch (Exception e) {

        }
    }

    public boolean isPatreon(EntityPlayer player) {
        return player != null && player.getUniqueID() != null && idList.contains(player.getUniqueID().toString());
    }
}

File: src/main/kotlin/net/lomeli/lomlib/client/layer/patreon/PatreonList.kt
package net.lomeli.lomlib.client.layer.patreon

import com.google.common.collect.Lists
import com.google.gson.Gson
import net.lomeli.lomlib.LomLib

import java.io.InputStreamReader
import java.net.URL

import net.minecraft.entity.player.EntityPlayer

import net.lomeli.lomlib.lib.ModLibs

class PatreonList : Runnable {
    private val idList: MutableList<String>

    init {
        idList = Lists.newArrayList<String>()
    }

    override fun run() {
        getLatestList()
    }

    fun getLatestList() {
        try {
            idList.clear()
            val gson = Gson()
            val file = InputStreamReader(URL(ModLibs.PATREON_URL).openStream())
            val json = gson.fromJson(file, Array<String>::class.java)
            file.close()
            if (json != null) {
                for (s in json) {
                    idList.add(s)
                }
            }
        } catch (e: Exception) {

        }
    }

    fun isPatreon(player: EntityPlayer?): Boolean = player != null && player.persistentID != null && idList.contains(player.persistentID.toString())
}
